### ISSUE TITLE
Improve navigation to/from kata translations view

### ISSUE THREAD

**Issue by hobovsky** [codewars/codewars.com#1991]
I am always frustrated when I have to navigate from a kata page to its translations view, and (especially) back.

I would like to have the translations view incorporated into kata page (now, without the tabs bar and easy navigation means, translations sub-page seems to be completely separate view), link to this view put into kata page tabs, and to be able to freely switch between Details, Discourse, Forks, Solutions and Translations.

As an alternative, I considered adding such tab by my userscript #1758  (which I will do, some day), but I think it would be nice to have on CW website anyway.

Some screenshots how I would see it:

![image](https://user-images.githubusercontent.com/23709795/71098033-5065d380-21b1-11ea-8579-22893beaa28d.png)

![image](https://user-images.githubusercontent.com/23709795/71098081-670c2a80-21b1-11ea-9cf7-e5c0eaaf6fe5.png)

**Comment by Steffan153** [comment]
Looks like this has been fixed.

**Comment by hobovsky** [comment]
You probably use my user script, because I put this in there. It's not available directly on CodeWars.
Thanks for using the script! ;)

**Comment by hobovsky** [comment]
While we are at it, navigation in other areas of translations could also use some improvements too:
- it's impossible to navigate from translation editor to translation overview. After clicking "Save", you are stuck.
- it's impossible to navigate from translation editor to kata page. Clicking on kata title brings you to the translations page, which is quite separate from the main kata page (see original post).
- while viewing a translation overview, "Related kata" link leads to forks tab, which is probably the most useless tab on kata page, and hardly ever a desired target of navigation.

**Comment by kazk** [comment]
Made some quick changes, and will deploy it tomorrow.

- Navigating to the list of translations and back
  - Added tabs as described
- Navigating to the translation overview
  - After saving a translation, the title of the title becomes a link
- Navigating to the kata from translation editor and overview
  - Changed the link to the main kata view.

(I don't know the use case of the forks page, so I might get rid of it. If users come up with an interesting fork, they can share it by commenting on the original solution and linking to it. Listing all forks doesn't help with finding them.)

**Comment by kazk** [comment]
Deployed

**Comment by Blind4Basics** [comment]
> (I don't know the use case of the forks page, so I might get rid of it. If users come up with an interesting fork, they can share it by commenting on the original solution and linking to it. Listing all forks doesn't help with finding them.)

Actually, there arent many users playing with forks, so if another user want's to find forks as in `I may find interesting stuff right there`, it's far easier to walk through the forks page then the soluitons page (where forks are very hard to find).
But yeah, I'm not sure that many users use it this way...

**Comment by kazk** [comment]
Hmm, it's probably better to have a better filter on solutions page, then. I'll keep the forks page for now.

**Comment by Blind4Basics** [comment]
yeah, would be even better than the forks page.

(while you're at it, maybe adding https://github.com/codewars/issues/issues/75 ? ;o )

**Comment by kazk** [comment]
Most of the enhancements to solutions page is blocked by https://github.com/codewars/codewars.com/issues/2263#issuecomment-719997658

I'll try working on it this quarater.

**Comment by hobovsky** [comment]
One use case I had in mind is still not covered:
- I open up a translation to review it
- I want to train on it, so I click `Fork` and mess with the code, review assertion messages, whatever
- I find an issue, so I want to get back to the translation to post on its discussion (preferably in a new tab, to keep training further on the fork)
- Nowhere to click to navigate to the translation, _and_ to keep the training fork open

This issue could be alleviated a bit by allowing to fork translations into a new tab, but `FORK SELECTED` does not seem to respond to middle click. Translation can be forked only in the current tab.

**Comment by kazk** [comment]
I can add the link to the parent somewhere.

> Translation can be forked only in the current tab.

I think I've always used the "Fork" button.

**Comment by hobovsky** [comment]
> I think I've always used the "Fork" button.

Oh and for all these years I have never even noticed the blue "Fork" button aside of "Approve" and "Reject" ones. I am an idiot.

Scratch my last remark then. Now I know where to click ;) And since it _is_ possible to navigate from one view to another and keep both open, all this point becomes less of an issue.

**Comment by Blind4Basics** [comment]
not that, if ytou fork the current translation, you lose elligibility, iirc (at least in that language). That's why unnamed always use kumite to review translations instead of forking them, I believe.

**Comment by kazk** [comment]
:man_facepalming: I really need to look into this mess.
